File: examples/qdrant/python/quickstart.py
"""
Embex Quick Start - Qdrant Provider

Requires: Qdrant server running (docker run -p 6333:6333 qdrant/qdrant)
Run: python examples/qdrant/python/quickstart.py
"""
import asyncio
from embex import EmbexClient, Point


async def main():
    print("🚀 Embex Quick Start - Qdrant Provider\n")
    print("📋 Prerequisites: Qdrant server running at http://localhost:6333")
    print("   Start with: docker run -p 6333:6333 qdrant/qdrant\n")
    
    # Qdrant - requires server running
    client = EmbexClient(provider="qdrant", url="http://localhost:6333")
    
    collection_name = "documents"
    collection = client.collection(collection_name)
    
    # Clean up if exists
    try:
        await collection.delete_collection()
        print(f"✅ Cleaned up existing collection: {collection_name}")
    except Exception:
        pass
    
    # 1. Create Collection
    print(f"\n📦 Creating collection: {collection_name}")
    await collection.create(dimension=768, distance="cosine")
    print("   ✅ Collection created!")
    
    # 2. Insert Data
    print("\n📝 Inserting documents...")
    points = [
        Point(id="1", vector=[0.1] * 768, metadata={"title": "Hello World"}),
        Point(id="2", vector=[0.2] * 768, metadata={"title": "Embex is Fast"}),
        Point(id="3", vector=[0.15] * 768, metadata={"title": "Rust Powered"}),
    ]
    await collection.insert(points)
    print(f"   ✅ Inserted {len(points)} documents!")
    
    # 3. Search
    print("\n🔍 Searching...")
    results = await collection.search(vector=[0.12] * 768, top_k=2)
    print(f"   ✅ Found {len(results.results)} results:\n")
    for i, result in enumerate(results.results, 1):
        print(f"   {i}. {result.metadata.get('title', 'N/A')} (Score: {result.score:.4f})")
    
    print("\n🎉 Qdrant quick start complete!")
    print("\n💡 Next: Try examples/python/semantic_search.py for a real-world example")


if __name__ == "__main__":
    asyncio.run(main())

File: examples/qdrant/node/quickstart.ts
/**
 * Embex Quick Start - Qdrant Provider
 *
 * Requires: Qdrant server running (docker run -p 6333:6333 qdrant/qdrant)
 * Run: npx tsx examples/qdrant/node/quickstart.ts
 */

import { EmbexClient } from "@bridgerust/embex";

async function main() {
  console.log("🚀 Embex Quick Start - Qdrant Provider\n");
  console.log(
    "📋 Prerequisites: Qdrant server running at http://localhost:6333"
  );
  console.log("   Start with: docker run -p 6333:6333 qdrant/qdrant\n");

  // Qdrant - requires server running
  const client = new EmbexClient("qdrant", "http://localhost:6333");

  const collectionName = "documents";
  const collection = client.collection(collectionName);

  // Clean up if exists
  try {
    await collection.deleteCollection();
    console.log(`✅ Cleaned up existing collection: ${collectionName}`);
  } catch (e) {
    // Collection doesn't exist
  }

  // 1. Create Collection
  console.log(`\n📦 Creating collection: ${collectionName}`);
  await collection.create(768, "cosine");
  console.log("   ✅ Collection created!");

  // 2. Insert Data
  console.log("\n📝 Inserting documents...");
  const points = [
    {
      id: "1",
      vector: Array(768).fill(0.1),
      metadata: { title: "Hello World" },
    },
    {
      id: "2",
      vector: Array(768).fill(0.2),
      metadata: { title: "Embex is Fast" },
    },
    {
      id: "3",
      vector: Array(768).fill(0.15),
      metadata: { title: "Rust Powered" },
    },
  ];
  await collection.insert(points);
  console.log(`   ✅ Inserted ${points.length} documents!`);

  // 3. Search
  console.log("\n🔍 Searching...");
  const results = await collection.search(Array(768).fill(0.12), 2);
  console.log(`   ✅ Found ${results.results.length} results:\n`);
  results.results.forEach((result, i) => {
    console.log(
      `   ${i + 1}. ${
        result.metadata?.title || "N/A"
      } (Score: ${result.score.toFixed(4)})`
    );
  });

  console.log("\n🎉 Qdrant quick start complete!");
  console.log(
    "\n💡 Next: Try examples/node/rag_system.ts for a real-world example"
  );
}

main().catch(console.error);
